### PR TITLE
fix: improve missing native binding error message for arm64 musl

### PR DESCRIPTION
**Pull request by renaudgenard**
## Summary
- report the expected native package and local fallback when Lightning CSS cannot load a platform binding
- include the original package and fallback load errors so arm64 musl installs point at `lightningcss-linux-arm64-musl` instead of only `../lightningcss.linux-arm64-musl.node`
- add a loader test that simulates a missing Linux arm64 musl optional dependency

Refs parcel-bundler/lightningcss#1269.

## Test plan
- `node --check node/index.js`
- `node --check node/test/native-binding.test.mjs`
- `npm exec -- uvu node/test native-binding.test.mjs`

### CODE REVIEW

**Review by kdy1 — COMMENTED** [review]
I think this is a change that's pretty useful for edge cases, but the PR title is misleading.

File: node/test/native-binding.test.mjs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import fs from 'node:fs';
+import path from 'node:path';
+import vm from 'node:vm';
+import {fileURLToPath} from 'node:url';
+
+const dirname = path.dirname(fileURLToPath(import.meta.url));
+const indexPath = path.join(dirname, '..', 'index.js');
+const indexSource = fs.readFileSync(indexPath, 'utf8');
+
+function loadWithMissingNativeBinding(platform, arch, family) {
```
Comment (kdy1) [inline node/test/native-binding.test.mjs:12]: Also, I really don't like this test

**Comment by renaudgenard** [comment]
Thanks @kdy1, renamed the PR for clarity.
